extern/boostd-data/client/client.go

```diff
@@ -27,6 +28,9 @@ type Store struct {
 		MarkIndexErrored          func(context.Context, cid.Cid, error) error
 		IndexedAt                 func(context.Context, cid.Cid) (time.Time, error)
 		PiecesContainingMultihash func(context.Context, mh.Multihash) ([]cid.Cid, error)
+		RemoveDealForPiece        func(context.Context, cid.Cid, uuid.UUID) error
+		RemovePieceMetadata       func(context.Context, cid.Cid) error
```

Comment:
I'm not sure that we need the RemovePieceMetadata method - normally we should only ever need to call RemoveDealForPiece.
RemoveIndexes should only be called if something went wrong and the database got into an inconsistent state where we have indexes but not piece information, right?

Reply:
I kept all 3 removal method to cover edge cases where one or more DB operations fail.
Example:
Remove last deal -> remove metadata(fails) -> remove indexes (succeeds)
So now we can remove metadata as indexes are already gone but metadata itself was left.